src/Image.h

```diff
@@ -157,7 +158,8 @@ class ImageBase {
     /** Get the address of a particular pixel. */
     void *address_of(int x, int y = 0, int z = 0, int w = 0) const {
         uint8_t *ptr = (uint8_t *)origin;
-        size_t offset = x*stride_0 + y*stride_1 + z*stride_2 + w*stride_3;
+        size_t offset = (ssize_t)x*stride_0 + (ssize_t)y*stride_1 +
```

Comment:
I can't imagine ssize_t is the type that is wanted here. If a signed type the same size as size_t is needed, ptrdiff_t is the right choice.